src/RandNumGenerator.java

```diff
@@ -10,27 +10,56 @@ public static void main (String[] args) {
 
         //welcome message
         System.out.println("Welcome to Ruth's Simple Random Number Generator!");
+        System.out.println("Note, your bounds will be included in the range");
 
         //getting lower bound of range
-        System.out.println("Please input the lower bound of your range (i.e. 3)");
-        int lowerNum = mrScanner.nextInt();
-        System.out.println("The lower bound is: " + lowerNum);
-
-        //getting upper bound of range
-        System.out.println("Please input the upper bound of your range (i.e. 10)");
-        int upperNum = mrScanner.nextInt();
-        System.out.println("The upper bound is: " + lowerNum);
-
-        //getting amount of numbers
-        System.out.println("How many random numbers would you like?");
-        int amtNum = mrScanner.nextInt();
-        System.out.println("Generating " + amtNum + " number(s):");
-
-        int i = amtNum;
-        while (i > 0) {
-            int randNum = lowerNum + rand.nextInt(upperNum - lowerNum);
-            System.out.println(randNum);
-            i--;
+
+        boolean neg = true;
+        int lowerNum;
+        while (neg == true) {
+            System.out.println("Please input the lower bound of your range (i.e. 3)");
+            lowerNum = mrScanner.nextInt();
+            if (lowerNum < 0) {
+                System.out.println("You entered a negative number, please input a valid lower bound.");
+            }
+            else {
+                neg = false;
+                System.out.println("The lower bound is: " + lowerNum);
+
+                //getting upper bound of range
+                boolean invalNum = true;
+                int upperNum;
+                while (invalNum == true) {
+                    System.out.println("Please input the upper bound of your range (i.e. 10)");
+                    upperNum = mrScanner.nextInt();
+                    if (upperNum < lowerNum || upperNum == lowerNum) {
```

Comment:
good one !
more optimized version can be `upperNum <= lowerNum` instead of `upperNum < lowerNum || upperNum == lowerNum`
But things are good enough now